// Copyright 2019 The GraphicsFuzz Project Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package gfauto;

import "gfauto/common.proto";
import "gfauto/device.proto";


// A fuzzing session requires a Settings proto stored in "settings.json".
// Default values are defined in settings_util.py.
message Settings {
  // A list of devices, including the list of active devices names that will actually be used in this session.
  DeviceList device_list = 1;

  // A list of custom binaries; this should rarely be set, but can be used, for example, if you want to override the
  // default versions of some binaries for all devices. It is probably more common to override binaries for certain
  // devices, or to just update the hardcoded set of default binaries (in binaries_util.py) to their newer versions.
  repeated Binary custom_binaries = 2;

  // Allow at most this many duplicate crashes per device.
  // If 0 (the protobuf default value), a different default value will actually be used, as determined by gfauto.
  uint32 maximum_duplicate_crashes = 3;

  // Store at most this many fuzzing failure files.
  uint32 maximum_fuzz_failures = 4;
}
